atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultSweepAssignedBucketStore.java

```diff
@@ -81,6 +97,83 @@ public BucketStateAndIdentifier getBucketStateAndIdentifier() {
+
+        byte[] serializedBucketProgress = bucketIdentifierPersister.trySerialize(newStartingBucket.bucketIdentifier());
+        for (int attempt = 0; attempt < CAS_ATTEMPT_LIMIT; attempt++) {
+            try {
+                Optional<Long> currentStartingBucket = readCell(cell, bucketIdentifierPersister::tryDeserialize);
+                if (currentStartingBucket.isEmpty()) {
+                    keyValueService.checkAndSet(CheckAndSetRequest.newCell(TABLE_REF, cell, serializedBucketProgress));
+                    if (log.isDebugEnabled()) {
+                        log.debug("Persisted new starting bucket", SafeArg.of("newStartingBucket", newStartingBucket));
+                    }
+                } else {
+                    if (newStartingBucket.bucketIdentifier() > currentStartingBucket.get()) {
+                        keyValueService.checkAndSet(CheckAndSetRequest.singleCell(
+                                TABLE_REF,
+                                cell,
+                                bucketIdentifierPersister.trySerialize(currentStartingBucket.get()),
+                                bucketIdentifierPersister.trySerialize(newStartingBucket.bucketIdentifier())));
+                        if (log.isDebugEnabled()) {
+                            log.debug(
+                                    "Updated sweep bucket progress",
+                                    SafeArg.of("previousStartingBucket", currentStartingBucket),
+                                    SafeArg.of("newStartingBucket", newStartingBucket));
+                        }
+                    } else {
+                        log.info(
+                                "Attempted to update starting bucket, but the existing starting bucket in the database"
+                                        + " was already ahead of us (possible timelock lost lock?)",
+                                SafeArg.of("previousStartingBucket", currentStartingBucket),
+                                SafeArg.of("newStartingBucket", newStartingBucket));
+                    }
+                }
+                return;
+            } catch (RuntimeException e) {
+                if (attempt == CAS_ATTEMPT_LIMIT - 1) {
+                    log.warn(
+                            "Repeatedly failed to update starting bucket as part of sweep; throwing, some work may"
+                                    + " need to be re-done.",
+                            SafeArg.of("attemptedNewStartingBucket", newStartingBucket),
+                            SafeArg.of("numAttempts", CAS_ATTEMPT_LIMIT),
+                            e);
+                    throw e;
+                } else {
+                    log.info(
+                            "Failed to read or update starting bucket as part of sweep. Retrying",
+                            SafeArg.of("newStartingBucket", newStartingBucket),
+                            SafeArg.of("attemptNumber", attempt + 1),
```

Comment:
nit: usually for this kind of situation I prefer to have my loop index begin at 1 and proceed to `<= max_attempts`

Reply:
I'll change the original code (https://github.com/palantir/atlasdb/blob/develop/atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/progress/DefaultBucketProgressStore.java#L122) to match in another PR!